packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -227,29 +226,17 @@ export class W3mModalBase extends LitElement {
     const isSwitchingNamespace = ChainController.state.isSwitchingNamespace
     const isPrevDisconnected = !CoreHelperUtil.getPlainAddress(this.caipAddress)
     const isNextConnected = CoreHelperUtil.getPlainAddress(caipAddress)
-    const sessions = await SIWXUtil.getAllSessions()
-    const isNextAuthenticated =
-      caipAddress && SIWXUtil.getSIWX()
-        ? sessions.some(
-            session =>
-              session.data.accountAddress === ParseUtil.parseCaipAddress(caipAddress)?.address
-          )
-        : true
 
     // When users decline SIWE signature, we should close the modal
     const isDisconnectedInSameNamespace = !isNextConnected && !isSwitchingNamespace
 
-    // If user is switching to another namespace and connected in that namespace, we should go back
-    const isSwitchingNamespaceAndConnected =
-      isSwitchingNamespace && isNextConnected && isNextAuthenticated
```

Comment:
`isSwitchingNamespace && isNextConnected` was the original condition so I think they should stay